### PR TITLE
Feature: alert metrics

### PR DESCRIPTION
**Pull request by stefanotorresi**
This PR adds an `alert` collector that implements the following metrics:

- `sap_alert_ha_check`
- `sap_alert_ha_failover_active`

It also enriches the boilerplate of the `collector` package by adding a new `RecordConcurrently` function that helps dealing with multiple concurrent metric recording functions within the same colleciton cycle, which was needed to avoid hitting SAPControl sequentially and slowing down too much a single scrape.

Some refactoring was also applied in the process of adding new webservice structures.

### CODE REVIEW

File: collector/alert/alert.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	var err error
+
+	err = collector.RecordConcurrently([]func(ch chan<- prometheus.Metric) error{
+		c.recordHAConfigChecks,
+		c.recordHAFailoverConfigChecks,
+		c.recordHAFailoverActive,
+	}, ch)
+
+	if err != nil {
+		log.Warn(err)
```
Comment (MalloZup) [inline collector/alert/alert.go]: we could here add more context to error

File: collector/default_collector.go [hunk 2]
```diff
@@ -58,19 +56,22 @@ func (c *DefaultCollector) makeMetric(name string, value float64, valueType prom
...
-		fileInfo, err := os.Stat(path)
-		if err != nil || os.IsNotExist(err) {
-			return errors.Errorf("'%s' does not exist", path)
-		}
-		if fileInfo.IsDir() {
-			return errors.Errorf("'%s' is a directory", path)
-		}
-		if (fileInfo.Mode() & 0111) == 0 {
-			return errors.Errorf("'%s' is not executable", path)
+// run multiple metric recording functions concurrently
```
Comment (MalloZup) [inline collector/default_collector.go]: do we need to add concurrency? We all know that it add a layer of complexity

Comment (stefanotorresi) [reply collector/default_collector.go]: the function is just 15 lines of code, it's not that complex, really, and it allows for much faster scrapes. The SAPControl web service can be veeery slow.

File: collector/alert/alert.go (hunk 1 above)
Comment (stefanotorresi) [reply collector/alert/alert.go]: good point!

File: collector/default_collector.go (hunk 2 above)
Comment (MalloZup) [reply collector/default_collector.go]: ok I will review this more accurately tomorrow. A part of this looks good pr. I only a bit code :grin:

**Pull request merged**